core/src/main/java/com/datastrato/gravitino/listener/info/TableInfo.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.info;
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.rel.Column;
+import com.datastrato.gravitino.rel.Table;
+import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
+import com.datastrato.gravitino.rel.expressions.distributions.Distributions;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
+import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.indexes.Index;
+import com.datastrato.gravitino.rel.indexes.Indexes;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * TableInfo exposes table interfaces for event listener, it's supposed to be read only. Most of the
+ * fields are shallow copied internally not deep copies for performance.
+ */
+public class TableInfo implements Table {
```

Comment:
It would be better to make this class `final`. Also I think we can directly expose the `TableInfo` to the users, not so necessary to use a `Table` interface. One thing I think is that we will add more interfaces to `Table`, but we may not need to expose this to `TableInfo`, inheriting `Table` will force us to expose everything when interface is changed.

Reply:
done